src/ert/_c_wrappers/enkf/model_config.py

```diff
@@ -1,299 +1,122 @@
-
-    def getRunpathAsString(self) -> str:
-        return self._get_runpath_as_char()
-
-    def selectRunpath(self, path_key) -> bool:
-        return self._select_runpath(path_key)
-
-    def setRunpath(self, path_format):
-        self._set_runpath(path_format)
-
-    def free(self):
-        self._free()
-
-    def getGenKWExportName(self) -> str:
-        return self._gen_kw_export_name()
-
-    def runpathRequiresIterations(self) -> bool:
-        return self._runpath_requires_iterations()
+        self.num_realizations = num_realizations
+        self.refcase = refcase
+        if config_dir is None:
+            config_dir = os.getcwd()
+        self._config_dir = config_dir
+        self.data_root = data_root if data_root is not None else config_dir
+        self.ens_path: str = (
+            ens_path
+            if ens_path is not None
+            else os.path.join(config_dir, self.DEFAULT_ENSPATH)
+        )
 
-    def getJobnameFormat(self) -> Optional[str]:
-        """Returns None if no job name format given in config file."""
-        return self._get_jobname_fmt()
+        self.history_source = (
+            history_source
+            if self.refcase is not None and history_source is not None
+            else self.DEFAULT_HISTORY_SOURCE
+        )
 
-    @property
-    def obs_config_file(self):
-        return self._get_obs_config_file()
+        self.runpath_format_string = (
+            runpath_format_string
+            if runpath_format_string is not None
+            else os.path.realpath(self.DEFAULT_RUNPATH)
+        )
 
-    def getEnspath(self) -> str:
-        return self._get_enspath()
+        self.jobname_format_string = jobname_format_string
+        self.gen_kw_export_name = (
+            gen_kw_export_name
+            if gen_kw_export_name is not None
+            else self.DEFAULT_GEN_KW_EXPORT_NAME
+        )
 
-    def getRunpathFormat(self) -> PathFormat:
-        return self._get_runpath_fmt()
+        self.obs_config_file = obs_config_file
 
-    @property
-    def num_realizations(self):
-        return self._get_num_realizations()
+        self.time_map = None
+        self._time_map_file = time_map_file
 
-    def data_root(self):
-        return self._get_data_root()
+        if time_map_file is not None:
+            self.time_map = TimeMap()
+            try:
+                self.time_map.fload(time_map_file)
+            except ValueError as err:
+                logger.warning(err)
+            except IOError as err:
+                logger.warning(f"failed to load timemap - {err}")
 
-    def get_time_map(self):
-        return self._get_time_map()
+    def get_last_history_restart(self) -> int:
+        if self.refcase:
+            return self.refcase.last_report
+        if self.time_map:
+            return self.time_map.last_step()
+        return -1
 
     def __repr__(self):
-        return f"<ModelConfig({self.__str__()})>"
+        return f"ModelConfig(\n{self.__str__()}\n)"
```

Comment:
This  `f"ModelConfig(\n{self.__str__()}\n)"` is equivalent with `f"ModelConfig(\n{self}\n)"`

Reply:
ah thank you, fixed!